Middleware docs (#5625)
vuestorefront/vue-storefront: packages/core/docs/guide/composables.md

In this example, we are using `useUiNotification` - a composable that handles notifications state. You can read more about it in the API reference.

[//]: # 'TODO: This should be added to API reference'

### How to customize graphql queries?

> If the integration you are currently using has GraphQL API, you may need to change the default query that is being sent to fetch the data. That's quite a common case so Vue Storefront also provides that ability.

Reviewer: ```suggestion
If your integration uses GraphQL API, you may need to change the default query that is being sent to fetch the data. That's quite a common case and Vue Storefront also provides the mechanism for this.
```

> Since the comminication with an API goes over the our middleware, all of the queries also are defined there.

Reviewer: ```suggestion
Since the communication with the API goes through our middleware, all queries also are defined there.
```

To customize or even totally override the original (default) queries you need to follow two steps.

Firstly, you need to use a dedicated parameter: `customQuery` that tells the app, what to do with a query.
This parameter is an object that has a name of the queries as keys, and the name of the queries function under the values.

```ts
const { search } = useProduct();

search({ customQuery: { products: 'my-products-query' } }); 
```

In the example above, we are changing `products` query, and our function that will take care of this overriding is `my-products-query`. As a second step, we need to define that function.

> Each definition of query function we can find in the `middleware.config.js`. Therefore, this is the place where we define `my-products-query`:

Reviewer: ```suggestion
Each custom query lives in the `middleware.config.js`, so it's the place where we should define `my-products-query`:
```

```js
module.exports = {
  integrations: {
    ct: {
      location: '@vue-storefront/commercetools-api/server',
      configuration: { /* ... */ },
      customQueries: {
        'my-products-query': ({ query, variables }) => {

          variables.locale = 'en'

          return { query, variables }
        }
      }
    }
  }
};
```

> The custom query function always has in the arguments the default query and default variables and must return the query and its variables as well. In the body you can do anything you want with those parameters - you can override them or even change to the new ones.

Reviewer: ```suggestion
The custom query function always has the default query and default variables in the arguments and must return the query and its variables. In the body you can do anything you want with those parameters - you can override them or even change to the new ones.
```
